sdk/v3/authn_client.go

```diff
@@ -1,12 +1,19 @@
 package sdk
 
 import (
+	"context"
+	"net/http"
+
+	rm "github.com/brigadecore/brigade/sdk/v3/internal/restmachinery"
 	"github.com/brigadecore/brigade/sdk/v3/restmachinery"
 )
 
 // AuthnClient is the root of a tree of more specialized API clients for dealing
 // with identity and authentication.
 type AuthnClient interface {
+	// WhoAmI returns a PrincipalReference for the currently authenticated
+	// principal.
+	WhoAmI(context.Context) (PrincipalReference, error)
```

Comment:
The authn client can return more specialized clients for users and service accounts, but I figured whoami was better to include right here. If a caller doesn't know who they are, they may also not know if they're a service account or a user.